go/enclave/enclave.go

```diff
@@ -569,28 +567,9 @@ func (e *enclaveImpl) EstimateGas(encryptedParams common.EncryptedParamsEstimate
 		return nil, fmt.Errorf("unable to decrypt params in EstimateGas request. Cause: %w", err)
 	}
 
-	// extract params from byte slice to array of strings
-	var paramList []interface{}
-	err = json.Unmarshal(paramBytes, &paramList)
-	if err != nil {
-		return nil, fmt.Errorf("unable to decode EstimateGas params - %w", err)
-	}
-
-	// params are [callMsg, block number (optional)]
-	// todo: handle blocknum, for now we always assume "latest" as with other methods
-	if len(paramList) == 0 {
-		return nil, fmt.Errorf("required at least one param, but received zero")
-	}
-
-	// convert the params[0] into an ethereum.CallMsg
-	callMsgBytes, err := json.Marshal(paramList[0])
-	if err != nil {
-		return nil, fmt.Errorf("unable to marshall callMsg - %w", err)
-	}
-	var callMsg ethereum.CallMsg
-	err = json.Unmarshal(callMsgBytes, &callMsg)
+	callMsg, _, err := rpc.ExtractEthCall(paramBytes)
```

Comment:
All calls that use Eth_Call https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_call reuse this method.